### ISSUE TITLE
Missing gif export

### ISSUE THREAD

**Issue by isomemo** [Pandora-IsoMemo/iso-app#107]
@arunge the option to export a plot as a gif from the models KernelTimeR or TimeR is now missing. Please re-introduce. 
Has anything else gone missing?

**Comment by arunge** [comment]
Ok, I will check what happened there tomorrow. There had not been any other
error messages. So we hope all is still working. In the worst case it
should be easy to downgrade the R version again, but than we must look for
a solution regarding rgpt3.

isomemo ***@***.***> schrieb am Do., 23. Feb. 2023, 18:46:

**Comment by arunge** [comment]
@isomemo Most likely, you mean the following export option (1)

![image](https://user-images.githubusercontent.com/16759098/221166398-e9c55392-825f-4ee9-9584-bb9f285ae983.png)

![image](https://user-images.githubusercontent.com/16759098/221167749-d65c71c8-5fb6-4a97-a486-dfd6cf35668b.png)

We had to remove the `gif` option several months ago because the calculation of a `gif` file sometimes was too slow. As an alternative, we added the feature to export a time series as a .zip (3), compere [this PR](https://github.com/Pandora-IsoMemo/iso-app/pull/58).

So nothing went lost here. :slightly_smiling_face:  Can I close this issue?

**Comment by isomemo** [comment]
@arunge I do not remember being informed of this or I misunderstood the issue. Users will not know how to generate the gif. Why was the export as gif too slow? Assuming that it was because all pictures were being merged at the same time, why not create a gif with just two slides initially and then incrementally add to this the remaining slides?

**Comment by isomemo** [comment]
@arunge a quick follow-up on this:

1. The same options should be re-implemented (e.g., timing between gif slides, reversing order or not)
2. The current export can be maintained but you would add to the zip file the animated gif optionally. The user will decide to include or not the gif and/or individual images. If only gif exported then no need for zip
3. Below an example of code where images are added to a gif file sequentially. This should overcome memory management issues:

#at the start the gif will only contain two images
# Load packages
library(magick)

# Store file names
files <- c("1.jpg", "2.jpg")

# Create animated gif
image_list <- lapply(files, image_read)
image_animate(image_list, loop = 0)
image_write(image_list, path = "animated.gif")

#after the above is done a loop can sequentially add new images. Below I illustrate only how to add one image

# Create image object for new slide
new_slide <- image_read("3.jpg")

# Read in existing gif
existing_gif <- image_read("animated.gif")

# Append new slide to existing gif
final_gif <- c(existing_gif, new_slide)

# Write new gif
image_write(final_gif, path = "animated.gif")

**Comment by isomemo** [comment]
@arunge I tried to generate a gif with many images (in the example you give above I set Time Interval to 10). The app crashed.

Perhaps it would be better to implement the algorithm as I suggested? Instead of trying to create the gif in a single function call create a loop and aggregate images one by one.

**Comment by arunge** [comment]
@isomemo  Ok, than I will try this. My hope was that it could work since we use now a different package to create the gif.
However, I will implement the loop and let you know when you can test again. Thanks for your patience!

**Comment by arunge** [comment]
@isomemo There was another issue here. The function `image_read()` that you suggested for reading image files did fail when trying to read a pdf file. There is a [recommendation](https://www.rdocumentation.org/packages/magick/versions/2.7.4/topics/editing) to use `image_read_svg()` and `image_read_pdf()` for svg and pdf files, respectively. To implemenent this for svg this was not an issue. To use `image_read_pdf()` the R package `pdftools` is needed. Currently the installation fails because of some missing packages under linux.

Should we have a look for that and try to fix the installation? Another option would be to exclude the export of gifs if the file type `pdf` is selected.

**Comment by isomemo** [comment]
@arunge please fix the problem using the technical options you find best. However, I do not understand the comment. Why is the creation of pdf 's relevant to creating the gif for export? I assume that the map has different export options and I assume that there you can choose the best ones for gif export.

**Comment by arunge** [comment]
> Why is the creation of pdf 's relevant to creating the gif for export?

@isomemo Currently, the file type that is selected by the user is used to create the files that go into the `.zip` file and/or are used to generate the `.gif` file. But of course, if it makes no difference, I can always generate the `.gif` file from e.g. `.png` files. Then, there is no need to add other R packages.

**Comment by isomemo** [comment]
@arunge well, I do not know how the code is implemented but to create a gif you need raster inputs (whatever format is easier/faster). In parallel the zip can contain the pdfs if the user selects pdf as output but these are obviously not used to create the gif. In summary, we are talking about two different things.

**Comment by arunge** [comment]
@isomemo With version 23.06.3 on [iso-app beta](https://isomemoapp.com/app/iso-memo-app-beta) the gif export is now also working when "pdf" is selected. 

Now, I always create jpeg files first (they are faster than png), and from them I generate the gif. No additional packages are required.

Feel free to test and in case of issues let me know. I am closing here.
